File: src/main/java/com/group/educate/educate/models/Roadmap/RoadmapModule.java
//Created by Ziad on 30/10/2025

package com.group.educate.educate.models.Roadmap;

import com.group.educate.educate.models.Roadmap.Skills.Skills;

import java.util.ArrayList;
import java.util.List;


public class RoadmapModule {

    private static int counter=0;
    private final int moduleId;
    private String name;
    private String description;

    private ArrayList<Skills> skills = new ArrayList<>();

    public RoadmapModule(String name, String description) {
        this.moduleId=++counter;
        this.name = name;
        this.description = description;
    }

    public RoadmapModule(String name, String description, Skills[] skills) {
        this(name, description);
        addSkills(skills);
    }

    public int getModuleID() {
        return moduleId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void addSkills(Skills... skills) {
        this.skills.addAll(List.of(skills));
    }
}

File: Frontend/app/Model/Roadmap/RoadmapModule.ts
import type {Skill} from "~/Model/Roadmap/Skill";

export class RoadmapModule {

    id: bigint;
    name: String;
    description: String;

    allSkills: Skill[];

    constructor(id: bigint, name: String, description: String, skills: Skill[]) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.allSkills = skills;
    }
}